hack/packages/package-tools/cmd/package-bundle-push.go

```diff
@@ -122,34 +123,40 @@ func validatePackageBundlePushFlags() error {
 	return nil
 }
 
-// prunePackages will update the given repository packages list to contain only
-// the bundle packages that match the first argument which contains a
-// comma-separated list of package bundles. If no package bundles are provided
-// or one cannot be found, an error is returned.
-func prunePackages(repository *Repository, args []string) error {
-	if len(args) == 0 {
-		return fmt.Errorf("at least one package bundle name is required to be specified")
-	}
-
-	// Only the first argument of the command will be recognized. The argument
-	// is expected to contain a list of comma separated package bundles.
-	csvBundles := args[0]
+// prunePackages will update in place the given map of repository packages to
+// contain only the bundle packages listed in csvBundles.
+func prunePackages(repos map[string]Repository, csvBundles string) error {
 	bundles := strings.Split(csvBundles, ",")
 
-	var pruned []Package
+	prunedPkgs := make(map[string][]Package)
+	for repoName := range repos {
+		prunedPkgs[repoName] = []Package{}
+	}
+
 	for _, bundle := range bundles {
```

Comment:
I don't think improving the performance (like you said, we're looping through just two repos) is worth the change, but perhaps readability is. However, I think it's not _that_ hard to read the nested for loops.

I'll let this one go (I won't refactor code to remove a for loop), mainly because at the moment, I'm not sure how even the new data structure you proposed would help. I could be wrong, but even with the new data structure, you need to know the repo name don't you? Not knowing the repo name, e.g. "management", then you'll still be forced to loop through all the repositories.